ex22_ch4.py

amymhaddad opened the pull request:
Write a program that asks for three numbers. Check first to see that all numbers are different. If they’re not different, then exit the program. Otherwise, display the largest number of the three.

amymhaddad commented on ex22_ch4.py:
```diff
@@ -0,0 +1,36 @@
+number2 = int(input("Enter a second number: "))
+number3 = int(input("Enter a third number: "))
+
+#step2: set up inequalities to make sure all numbers are different. If they're not different, write a message that says that and exit the game.
```
For this section, I originally combined all of these "elifs" on one line. However, the program didn't run correctly consistently. Also, is it preferred to combine multiple inequality statements (see below) or put them on separate lines (as shown in file)?

if number1 == number2 and number1 == number3 and number2 == number1 and number2 == number3 and number3 == number1 and number3 == number2:
    print("The numbers are not different.")
else: 
    print("Let's continue with the game.")

paulghaddad replied:
Yeah, you did the overall set up correctly to keep the check that all the numbers are different from the check to see which is largest. For step 2, I'd suggest:

```
if number1 != number2 or number1 != number3 or number2 != number3:
```

This will be `True` is any of the three comparisons fail.  In your example, ` number1 == number2` is the same as `number2 == number1`, for example. Also, you'd want an `OR`. Using `and`s instead would require all the comparisons to be `True`. However, we want to know if any of them are `True`, which means we want `or`.

paulghaddad commented on ex22_ch4.py:
```diff
@@ -0,0 +1,36 @@
+    print("Let's continue with the game.")
+
+#step3: set up conditional to see which number is the largest
+if number1 > number2 and number1 > number3:
```
Perfect!

paulghaddad commented on ex22_ch4.py:
```diff
@@ -0,0 +1,36 @@
+#step3: set up conditional to see which number is the largest
+if number1 > number2 and number1 > number3:
+    print(f"{number1} is the largest number.")
+elif number2 > number1 and number2 > number3:
```
Perfect!

paulghaddad commented on ex22_ch4.py:
```diff
@@ -0,0 +1,36 @@
+    print(f"{number1} is the largest number.")
+elif number2 > number1 and number2 > number3:
+    print(f"{number2} is the largest number.")
+elif number3 > number2 and number3 > number1:
```
If lines 31 and 33 aren't `True` that means number3 must be the largest number, so you can just do an `else` and avoid the comparisons on line 35.

amymhaddad commented:
@paulghaddad I updated this exercise:
-The largest number was printing for all conditions, so I needed to re-write the conditional statements. Now the largest number only appears if the numbers are different.
-I updated the inequalities.

paulghaddad commented on ex22_ch4.py:
```diff
@@ -0,0 +1,23 @@
+number3 = int(input("Enter a third number: "))
+
+#step2: set up inequalities to make sure all numbers are different. Then, set up another conditional if the numbers are different.
+if number1 == number2 or number1 == number3 or number2 == number3:
```
Excellent.

paulghaddad commented on ex22_ch4.py:
```diff
@@ -0,0 +1,23 @@
+#step2: set up inequalities to make sure all numbers are different. Then, set up another conditional if the numbers are different.
+if number1 == number2 or number1 == number3 or number2 == number3:
+    print("The numbers are not different. Game over.")
+elif number1 != number2 or number1 != number3 or number2 != number3:
```
I think you can use an `else` here and not have to do the conditional. If the numbers are different (which you test for on line 14), then you know they are different and can just proceed with the else.

paulghaddad commented on ex22_ch4.py:
```diff
@@ -0,0 +1,23 @@
+        print(f"{number1} is the largest number.")
+    elif number2 > number1 and number2 > number3:
+        print(f"{number2} is the largest number.")
+    elif number3 > number2 and number3 > number1:
```
This can also be an `if`. If the conditionals on line 18 and 20 don't succeed, the only possible result is that number 3 is largest. So you can just use an `else` and not test for it.

amymhaddad commented:
@paulghaddad I agree with your suggestions to add "else". It works in both places. I made the updates.
